src/libraries/Common/src/System/IO/Hashing/XxHash64.cs

```diff
@@ -15,7 +15,12 @@ namespace System.IO.Hashing
     /// For methods that persist the computed numerical hash value as bytes,
     /// the value is written in the Big Endian byte order.
     /// </remarks>
-    public sealed partial class XxHash64 : NonCryptographicHashAlgorithm
+#if SYSTEM_PRIVATE_CORELIB
+    internal
```

Comment:
Why are we bringing in XxHash64?

Reply:
@stephentoub good point! I have removed all unrelated dependencies. The reason xxhash64 was referenced was its `Avalance` function.

It turned out we had two `ulong Avalance(ulong)` functions, one in shared another in XxHash64 and all implementations used both claiming that the shared one is faster when bits are known to be partially mixed.

I've moved the XxHash64 to shared too and renamed the existing fast one as `FastAvalanche`. 
Also, I split XxHash3 into two files: XxHash3 (static) and XxHash3.State.cs